### ISSUE TITLE
(Minor Bug) Lush Biomes Mod in Nether

### ISSUE THREAD

**Issue by walrusgumbo** [Glitchfiend/BiomesOPlenty#396]
![2015-01-28_12 46 11](https://cloud.githubusercontent.com/assets/10465957/5943594/4c306714-a6ee-11e4-9e48-b3f5d85d6dda.png)
So this happens.

**Comment by walrusgumbo** [comment]
Let me clarify: the Lush River Biome spawns in the Nether.

**Comment by Taschenmogul** [comment]
Just a more or less random comment - we have seen this problem before and it was supposedly fixed a while ago: https://github.com/Glitchfiend/BiomesOPlenty/issues/334
Several people, including me, have seen this emerge again though and I believe the reason is this, citing Adubbz:
> I think overwriting the GenLayer and adding a check to ensure the biome is an instance of BiomeGenMutated should do the trick.

If I understand that correctly then BiomeGenMutated is the class that is used by mutations of biomes that can occur in a "base" biome, also called sub-biomes.
So, if BOP now only generates biomes as sub-biomes within other biomes if they are instances of the class BiomeGenMutated then that would mean that biomes still can erroneously appear as sub-biomes within other biomes if they are indeed instances of the class BiomeGenMutated, that is, originally made to be sub-biomes.

If I was right the problem here would be that it is not necessarily easy to determine if a biome is implemented as a sub-biome or not. Your case makes it seem that Lush River is implemented as a sub-biome, the config files though do not show that and one would have to look into the source code to confirm if it is or not. Looking at the NEI biome dump I can see what class is used but I cannot see if that class extends BiomeGenMutated, so this basically goes for all other mods too.

### TL;DR:
Make a NEI biome dump to see the registered biomes and their IDs. Open up the resulting file (for CSV files use a spreadsheet software like OpenOffice Calc), look up the biome IDs of the Lush River biome and the nether biome in which it generated. If my theory is correct then they should be exactly 128 ID values appart.
In that case open up your config files and try to rearrange the IDs so that they do not conflict; if possible keep them all below 128. In my experience the four Thaumcraft biomes don't pose a problem, they can remain at IDs above 127, but also watch out for AppliedEnergistics's Storage Cell biome as I have already seen that generate as a sub-biome of another biome.

That said keep in mind that this method will not fix the problem in already generated chunks on your server, and changing biome IDs will change biome-generation for the seed you are using, likely leading to sharp transitions (have never really tried that so can't say for sure), so even if this works it would likely require you to start a new world or live with the skewed generation.
